tests/test_models.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+from tensorflow import keras
+
+from ms2deepscore.data_generators import DataGeneratorAllInchikeys
+from ms2deepscore.models import SiameseModel
+from tests.test_data_generators import create_test_data
+
+
+def get_test_generator():
+    # Get test data
+    spectrums_binned, score_array, inchikey_score_mapping, inchikeys_all = create_test_data()
+
+    dimension = 101
+    same_prob_bins = [(0, 0.5), (0.5, 1)]
+    inchikey_ids = list(np.arange(0, 80))
+
+    # Create generator
+    return DataGeneratorAllInchikeys(spectrums_binned, score_array, inchikey_ids,
+                                     inchikey_score_mapping, inchikeys_all,
+                                     dim=dimension, same_prob_bins=same_prob_bins)
+
+
+def test_siamese_model():
+    test_generator = get_test_generator()
+    model = SiameseModel(input_dim=101, base_dims=(200, 200, 200),
+                         embedding_dim=200, dropout_rate=0.2)
+    model.compile(loss='mse', optimizer=keras.optimizers.Adam(lr=0.001))
+    model.summary()
+    model.fit(test_generator,
+              validation_data=test_generator,
+              epochs=2)
```

Comment:
I am not very certain **what** to assert with such a model, so I just picked a few things that came to my mind. Feel free to change/adapt.

Reply:
I think it would be nice to further extent the test to also check the `.base` model!
Maybe add the following after the above mentioned asserts?
```python
# Test base model ("encoder")
X, y = test_generator.__getitem__(0)
embeddings = model.base.predict(X[0])
assert isinstance(embeddings, np.ndarray), "Expected numpy array"
assert embeddings.shape[0] == test_generator.batch_size == 32, "Expected different batch size"
assert embeddings.shape[1] == model.base.output_shape[1] == 200, "Expected different embedding size"
```

Reply:
I think these assertions make a lot of sense! I also added the ones for inference using the base model.